service/assertion/actions.go

```diff
@@ -102,16 +102,42 @@ func CreateModelAssertionHeaders(m datastore.Model) (map[string]interface{}, dat
 		"brand-id":          m.BrandID,
 		"series":            fmt.Sprintf("%d", assert.Series),
 		"model":             m.Name,
-		"architecture":      assert.Architecture,
 		"store":             assert.Store,
-		"gadget":            assert.Gadget,
-		"kernel":            assert.Kernel,
 		"sign-key-sha3-384": keypair.KeyID,
 		"timestamp":         time.Now().Format(time.RFC3339),
 	}
 
-	// Check if the optional required-snaps field is needed
-	if len(assert.RequiredSnaps) == 0 {
+	// Add the optional fields as needed
+	if len(assert.Classic) != 0 {
+		headers["classic"] = assert.Classic
+	}
+
+	if len(assert.DisplayName) != 0 {
+		headers["display-name"] = assert.DisplayName
+	}
+
+	// Some headers are required for Ubuntu Core, whilst optional or invalid for Classic
+	if headers["classic"] == "true" {
```

Comment:
Is validated somehow the case that the model creator wanted to set this as true but wrote "True" or similar?. It would be bad that it goes to false because of a misspelling.
If not, you could use a switch here to evaluate "true", "false" and default

Reply:
It's done in the UI - attempts to correct the typos. But it could be called from the API, so I'll add an additional check here.